engine/lib/notification.php

```diff
@@ -577,6 +577,35 @@ function set_user_notification_setting($user_guid, $method, $value) {
 	return false;
 }
 
+
+/**
+ * Takes a string of the form "Evan <evan@elgg.org>" and turns it into an array
+ * of the form: ['name' => 'Evan', 'email' => 'evan@elgg.org'].
+ * 
+ * This is not very sophisticated and only used to provide a light BC effort.
+ * 
+ * In the future we should just prefer building the Swift_Message directly...
+ * 
+ * @param string $contact e.g. "Evan <evan@elgg.org>"
+ * 
+ * @return array e.g. ['name' => 'Evan', 'email' => 'evan@elgg.org']
+ * @access private
+ */
+function _elgg_parse_email_contact($contact) {
+	$containsName = preg_match('/<(.*)>/', $contact, $matches) == 1;
+	if ($containsName) {
+		return [
+			'name' => trim(elgg_substr($contact, 0, $email_pos)),
```

Comment:
Where does $email_pos come from? If $containsName, just use the same pattern with preg_replace to remove it and trim the result?